api/src/main/java/org/openmrs/ui/framework/page/PageContext.java

```diff
@@ -189,7 +190,18 @@ public void setLocale(Locale locale) {
 
     @Override
     public void includeResource(Resource resource) {
-        resourcesToInclude.add(resource);
+    	Iterator<Resource> iterator = resourcesToInclude.iterator();
+    	while (iterator.hasNext()) {	
+    		Resource res = iterator.next();
+    	    if (res.getProviderName().equals(resource.getProviderName()) && res.getResourcePath().equals(resource.getResourcePath())) {
+    	    	if (res.getPriority() < resource.getPriority()) {
+    	    		iterator.remove();
+    	    		resourcesToInclude.add(resource);
+    	    	}
+    			return;
+    		} 
+    	}
+    	resourcesToInclude.add(resource);   		 

```

Comment:
Yes, just messed up with the nex line indentation (L205). Please make sure that line is proper indented.

Reply:
Since this is the non returnable funtion (void), we are not following to add return to breck the iteration.  Why don't you use the `break`?

Reply:
Don't you think you wanna to give a second glance to this function. What will actually happen if I just break from the loop?

Reply:
So what happens if the Line201 make a return? The same thing will happen for the break also

Reply:
Well if you use <code>break</code>, it will just break from the loop and and also executing line204 which we don't intend to at that particular condition. It may include duplicate resources.

Reply:
yes, So what do you think about this, 
1. L199 Removed
2. L201 changed to `break`
```
    	Iterator<Resource> iterator = resourcesToInclude.iterator();
    	while (iterator.hasNext()) {
    	    Resource res = iterator.next();
    	    if (res.getProviderName().equals(resource.getProviderName()) && res.getResourcePath().equals(resource.getResourcePath())) {
    	    	if (res.getPriority() < resource.getPriority()) {
    	    		iterator.remove();
    	    	}
    	    	break;
    	    } 
    	}
    	resourcesToInclude.add(resource); 
```

Reply:
I think that won't work. I will include resources of lower priority if those of higher priority were already included. I personally think its of no harm to <code>return</code> from a <code>void</code> function.


<code> 

            Resource first = new Resource(Resource.CATEGORY_JS, "source", "z_first", 1);
                
            Resource second = new Resource(Resource.CATEGORY_JS, "mergepatientdata", "path", 5); // First include one of higher priority
               
            Resource duplicateResource = new Resource(Resource.CATEGORY_JS, "mergepatientdata", "path", 3); // Then include a duplicate with lower priority

		PageContext pageContext = new PageContext(null);

		pageContext.includeResource(first);

		pageContext.includeResource(second);

		pageContext.includeResource(duplicateResource);

		List<Resource> jsResourcesToIncluude = pageContext.getResourcesToInclude(Resource.CATEGORY_JS);
		assertEquals(2, jsResourcesToIncluude.size()); // This assertion fails. It's 3 instead of 2.
</code>